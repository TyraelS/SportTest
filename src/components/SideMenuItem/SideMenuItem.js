import styled from 'styled-components';

const getSideMenuStyles = props =>
  props.active
    ? `background-color: #bebebe;
  color: #fff;`
    : `background-color: #cecece;
  color: #fff;`;

const SideMenuItem = styled.div`
  cursor: pointer;
  width: 300px;
  height: 50px;
  line-height: 50px;
  text-align: center;
  ${getSideMenuStyles}
`;

export default SideMenuItem;
