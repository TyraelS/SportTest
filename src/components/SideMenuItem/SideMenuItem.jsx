import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import SideMenuItemWrapper from './SideMenuItemWrapper.style';

export const sideMenuItemClick = (id, sportItemClick) => () => {
  sportItemClick(id);
};

export default class SideMenuItem extends PureComponent {
  render() {
    const { id, handler, active, children } = this.props;
    return (
      <SideMenuItemWrapper
        onClick={sideMenuItemClick(id, handler)}
        active={active}
      >
        {children}
      </SideMenuItemWrapper>
    );
  }
}

SideMenuItem.displayName = 'SideMenuItem';
SideMenuItem.propTypes = {
  id: PropTypes.string.isRequired,
  handler: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired
};
