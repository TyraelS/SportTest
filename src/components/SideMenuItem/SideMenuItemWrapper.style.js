import styled from 'styled-components';

const SideMenuItem = styled.div`
  cursor: pointer;
  width: 100%;
  height: 50px;
  line-height: 50px;
  text-align: center;
  color: ${props => props.theme.textSports};
  background-color: ${props =>
    props.active ? props.theme.accent : props.theme.regular};
`;

export default SideMenuItem;
