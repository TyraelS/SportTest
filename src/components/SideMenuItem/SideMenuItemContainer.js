import { compose, setDisplayName, pure } from 'recompose';

import SideMenuItem from './SideMenuItem';

const enhance = compose(
  setDisplayName('SideMenuItemContainer'),
  pure
);

export default enhance(SideMenuItem);
