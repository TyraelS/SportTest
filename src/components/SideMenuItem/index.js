import SideMenuItemContainer from './SideMenuItemContainer';

export default SideMenuItemContainer;
