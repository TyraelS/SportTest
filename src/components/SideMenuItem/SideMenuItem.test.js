import React from 'react';
import { shallow } from 'enzyme';

import SideMenuItem, { sideMenuItemClick } from './SideMenuItem';

describe('Given the sideMenuItemClick function', () => {
  const mockFunction = jest.fn();
  it('should call passed function with provided id', () => {
    sideMenuItemClick('1', mockFunction)();
    expect(mockFunction).toHaveBeenCalledWith('1');
  });
});

describe('Given the SideMenuItem component', () => {
  describe('when the component is rendered', () => {
    const component = shallow(
      <SideMenuItem id='1' handler={jest.fn()} active={true} children='text' />
    );
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
