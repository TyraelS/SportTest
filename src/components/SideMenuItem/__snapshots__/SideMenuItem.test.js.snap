// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the SideMenuItem component when the component is rendered should match the snapshot 1`] = `
<ForwardRef(styled.div)
  active={true}
  onClick={[Function]}
>
  text
</ForwardRef(styled.div)>
`;
